src/ggml.c

```diff
@@ -7570,44 +7481,15 @@ GGML_API struct ggml_tensor * ggml_conv_1d(
         int                   s0,
         int                   p0,
         int                   d0) {
-    struct ggml_tensor * result = ggml_conv_1d_stage_0(ctx, a, b, s0, p0, d0);
-    result = ggml_conv_1d_stage_1(ctx, a, result);
+    struct ggml_tensor * result = ggml_im2col(ctx, a, b, s0, 0, p0, 0, d0, 0, false); // [N, OH, OW, IC * KH * KW]
+    result = ggml_reshape_3d(ctx, ggml_cont(ctx, ggml_transpose(ctx,
+            ggml_mul_mat(ctx,
+            ggml_reshape_2d(ctx, a, (a->ne[0] * a->ne[1]),  a->ne[2]),
+            ggml_reshape_2d(ctx, result, result->ne[0],  (result->ne[2] * result->ne[1]))))),
+            result->ne[1],  a->ne[2], result->ne[2]);
```

Comment:
```suggestion
    struct ggml_tensor * result = ggml_im2col(ctx, a, b, s0, 0, p0, 0, d0, 0, false); // [N, OH, OW, IC * KH * KW]
    result = ggml_reshape_3d(ctx,
            ggml_mul_mat(ctx,
            ggml_reshape_2d(ctx, a, (a->ne[0] * a->ne[1]),  a->ne[2]),
            ggml_reshape_2d(ctx, result, result->ne[0],  (result->ne[2] * result->ne[1]))))),
            a->ne[2], result->ne[1], result->ne[2]);
```

I still think we don't need this `ggml_cont()` and `ggml_transpose()` here.
Have to take a deeper look to see if I'm missing something